src/Knp/Component/Pager/Event/Subscriber/Sortable/ArraySubscriber.php

```diff
@@ -2,82 +2,93 @@
 
 class ArraySubscriber implements EventSubscriberInterface
 {
-
     /**
      * @var string the field used to sort current object array list
      */
-    private $currentSortingFieldGetter;
+    private $currentSortingField;
 
     /**
      * @var string the sorting direction
      */
     private $sortDirection;
 
-    public function items(ItemsEvent $event)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(PropertyAccessorInterface $accessor = null)
     {
-        if (is_array($event->target) && count($event->target) > 1)
-        {
-            if (isset($_GET[$event->options['sortFieldParameterName']])) {
-                $this->sortDirection = isset($_GET[$event->options['sortDirectionParameterName']]) && strtolower($_GET[$event->options['sortDirectionParameterName']]) === 'asc' ? 'asc' : 'desc';
-
-                // TODO add whitelist
-//                if (isset($event->options['sortFieldWhitelist'])) {
-//                    if (!in_array($_GET[$event->options['sortFieldParameterName']], $event->options['sortFieldWhitelist'])) {
-//                        throw new \UnexpectedValueException("Cannot sort by: [{$_GET[$event->options['sortFieldParameterName']]}] this field is not in whitelist");
-//                    }
-//                }
-
-                $sortFieldParameterName = explode('.', $_GET[$event->options['sortFieldParameterName']]);
-                if(isset($sortFieldParameterName[1])) {
-                    // Capitalize first letter in order to prepare getter construction
-                    $sortFieldName = ucfirst($sortFieldParameterName[1]);
-
-                    $this->currentSortingFieldGetter = "get{$sortFieldName}";
-
-                    // Getter detection
-                    $class = new ReflectionClass(get_class(current($event->target)));
-                    if($class->hasMethod($this->currentSortingFieldGetter)) {
-                        // Sort
-                        usort($event->target, array($this, "sort" . ucfirst($this->sortDirection)));
-                    }
-                }
-            }
+        if (!$accessor && class_exists('Symfony\Component\PropertyAccess\PropertyAccess')) {
+            $accessor = PropertyAccess::createPropertyAccessorBuilder()->enableMagicCall()->getPropertyAccessor();
         }
+
+        $this->propertyAccessor = $accessor;
     }
 
-    /**
-     * @param $object1 first object to compare
-     * @param $object2 second object to compare
-     */
-    private function sortAsc($object1, $object2)
+    public function items(ItemsEvent $event)
     {
-        $fieldValue1 = strtolower($object1->{$this->currentSortingFieldGetter}());
-        $fieldValue2 = strtolower($object2->{$this->currentSortingFieldGetter}());
-        if ($fieldValue1 == $fieldValue2) {
-            return 0;
+        if (!is_array($event->target) || !isset($_GET[$event->options['sortFieldParameterName']])) {
```

Comment:
Usort doesn't call the callback if array is empty. But my style is to purposely leave out such things from top of the function if rest of the body is cheap to execute, because if user misconfigured something else, he will not know about it until he passes non-empty array here E.g. if he enables sorting by a field which is missing from (enabled) whitelist, this subscriber will appear to be working correctly until some data appear there.